src/coreclr/jit/emitriscv64.cpp

```diff
@@ -1262,15 +1262,27 @@ void emitter::emitLoadImmediate(emitAttr size, regNumber reg, ssize_t imm)
 
     // And load remaining part part by batches of 11 bits size.
     INT32 remainingShift = msb - 30;
+
+    UINT32 shiftAccumulator = 0;
     while (remainingShift > 0)
     {
         UINT32 shift = remainingShift >= 11 ? 11 : remainingShift % 11;
-        emitIns_R_R_I(INS_slli, size, reg, reg, shift);
+        UINT32 mask  = 0x7ff >> (11 - shift);
+        remainingShift -= shift;
+        ssize_t low11 = (imm >> remainingShift) & mask;
+        shiftAccumulator += shift;
+
+        if (low11)
```

Comment:
While you're here, shouldn't line 1261 `emitIns_R_R_I(INS_addiw, size, reg, reg, high31 & 0xFFF);` be under an analogous condition? I think immediates like 0x123000 also emit an unnecessary addiw.

Reply:
Thank you. Updated.